### DISCUSSION TITLE
How to use Vitest with Git pre commit hook?

### DISCUSSION THREAD

**Discussion by littlesandra88** [vitest-dev/vitest#1626] [Q&A] [answered]
Dear all =)

I would like to create a new project based on [this](https://github.com/stemmlerjs/simple-typescript-starter) , where I use Vitest instead of Jest.

Based on `.eslintrc` is seams is it EsLint that is calling Jest.

**Question**

What is the recommended way to use Vitest with Git pre commit hooks?
Is it Vitest that calls EsLint? Or EsLint that calls Vitest?

Any packages or help will be much appreciated =)

Hugs,
Sandra =)

**Comment by Demivan** [comment, marked as answer]
`jest` in `.eslintrc` is for eslint rules. This is because jest uses global imports that eslint does not like.
If you don't have globals enabled with Vitest you do not need eslint rules.

This line runs `test` script as a pre-commit hook:
https://github.com/stemmlerjs/simple-typescript-starter/blob/3b1fb5311aa6bc9dc8dbbc7749c9d3022a43d897/package.json#L24 
You only need to change `test` script to use `vitest` and husky will run it.

**Reply by nickserv** [reply]
Note that instead of using Husky, you can put hooks in a `.hooks` directory using `git config core.hooksPath .hooks`.
